compiler/rustc_session/src/options.rs

```diff
@@ -946,15 +946,9 @@ mod parse {
 
         for option in v.split(',') {
             match option {
-                "no-branch" => {
-                    slot.branch = false;
-                    slot.mcdc = false;
-                }
-                "branch" => slot.branch = true,
-                "mcdc" => {
-                    slot.branch = true;
-                    slot.mcdc = true;
-                }
+                "block" => slot.level = CoverageLevel::Block,
+                "branch" => slot.level = CoverageLevel::Branch,
+                "mcdc" => slot.level = CoverageLevel::Mcdc,
                 _ => return false,
             }
         }
```

Comment:
Do we keep the for loop for future changes ? Otherwise, I think we should get rid of it, as it wouldn't make sense to give `-Z coverage-options=mcdc,block` to the compiler

Reply:
(Some examples of possible future options: https://github.com/rust-lang/rust/issues/124144#issuecomment-2078967483, #120013, and perhaps an option to not expand/emit zero-width spans.)